File: Example/AzureDataAndroidExample_Java/app/src/main/java/com/azure/mobile/azuredataandroidexample_java/Adapter/ViewHolderFactory.java
package com.azure.mobile.azuredataandroidexample_java.Adapter;

import android.view.View;

import java.lang.reflect.Constructor;
import java.lang.reflect.InvocationTargetException;

/**
 * Created by mww121 on 26/10/17.
 */

public class ViewHolderFactory {

    /**
     * Creates a retrofit service from an arbitrary class (clazz)
     * @param clazz Java interface of the retrofit service
     * @return retrofit service with defined endpoint
     */
    public static <T> Object create(final Class<T> clazz, View v) {
        try {
            Constructor[] cs = clazz.getConstructors();
            //Constructor c = clazz.getConstructor(Activity.class, View.class);
            cs[0].setAccessible(true);
            return cs[0].newInstance(v);
        }
        catch (IllegalAccessException | InstantiationException | InvocationTargetException ex) {
            ex.printStackTrace();
        }

        return null;
    }
}

File: Example/AzureDataAndroidExample/app/src/main/java/com/azure/mobile/azuredataandroidexample/viewholder/ViewHolderFactory.kt
package com.azure.mobile.azuredataandroidexample.viewholder

import android.support.v7.widget.RecyclerView
import android.view.View
import java.lang.reflect.InvocationTargetException

/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */

object ViewHolderFactory {

    /**
     * Creates a retrofit service from an arbitrary class (clazz)
     * @param clazz Java interface of the retrofit service
     * @return retrofit service with defined endpoint
     */
    fun <T> create(clazz: Class<T>, v: View): RecyclerView.ViewHolder? {
        try {
            val cs = clazz.constructors
            //Constructor c = clazz.getConstructor(Activity.class, View.class);
            cs[0].isAccessible = true
            return cs[0].newInstance(v) as RecyclerView.ViewHolder
        } catch (ex: IllegalAccessException) {
            ex.printStackTrace()
        } catch (ex: InstantiationException) {
            ex.printStackTrace()
        } catch (ex: InvocationTargetException) {
            ex.printStackTrace()
        }


        return null
    }
}